File: home-hub/frontend/src/api/services.ts
import type { JellyfinMappings, ServicesCatalog } from '../types/services'
import { apiJson } from './http'

export async function fetchServicesCatalog(): Promise<ServicesCatalog> {
  return apiJson('/api/services/catalog')
}

export async function fetchJellyfinMappings(): Promise<JellyfinMappings> {
  return apiJson('/api/services/jellyfin')
}

export function healthLabel(status?: string): string {
  if (status === 'up') return '正常'
  if (status === 'down') return '不可用'
  return '未知'
}

export function healthClass(status?: string): string {
  if (status === 'up') return 'ok'
  if (status === 'down') return 'down'
  return 'unknown'
}

File: home-hub/backend/routers/services.py
"""本地服务目录 API。"""

from __future__ import annotations

from typing import Any

from fastapi import APIRouter

from backend.services import registry

router = APIRouter(prefix="/api/services", tags=["services"])


@router.get("/catalog")
async def services_catalog() -> dict[str, Any]:
    catalog = await registry.build_catalog_async(with_health=True)
    return catalog


@router.get("/jellyfin")
async def jellyfin_mappings() -> dict[str, Any]:
    return await registry.load_jellyfin_mappings_async()


@router.get("/health-summary")
async def health_summary() -> dict[str, Any]:
    catalog = await registry.build_catalog_async(with_health=True)
    items: list[dict[str, Any]] = []
    for cat in catalog.get("categories", []):
        for item in cat.get("items", []):
            items.append(
                {
                    "id": item.get("id"),
                    "name": item.get("name"),
                    "health": item.get("health"),
                    "docker_running": item.get("docker_running"),
                }
            )
    up = sum(1 for x in items if x.get("health", {}).get("status") == "up")
    return {"total": len(items), "up": up, "items": items}
